src/Pim/Bundle/DataGridBundle/Extension/Sorter/Product/ValueSorter.php

```diff
@@ -14,11 +15,34 @@
  */
 class ValueSorter implements SorterInterface
 {
+    /** @var AttributeRepositoryInterface */
+    protected $attributeRepository;
+
+    /**
+     * @param AttributeRepositoryInterface $attributeRepository
+     */
+    public function __construct(AttributeRepositoryInterface $attributeRepository)
+    {
+        $this->attributeRepository = $attributeRepository;
+    }
+
     /**
      * {@inheritdoc}
      */
     public function apply(DatasourceInterface $datasource, $field, $direction)
     {
-        $datasource->getProductQueryBuilder()->addSorter($field, $direction);
+        $attribute = $this->attributeRepository->findOneByIdentifier($field);
+
+        $context = [];
+        if (null !== $attribute) {
+            if (!$attribute->isScopable()) {
+                $context['scope'] = null;
+            }
+            if (!$attribute->isLocalizable()) {
```

Comment:
It comes from the datagrid context. In fact, it is always present, that's why I need to do that: so the datagrid context is overridden by this one when needed.